docs/asciidoc/static/complex-pipeline.asciidoc

```diff
@@ -0,0 +1,363 @@
+      "@version":"1",
+      "@timestamp":"2015-07-30T20:30:41.265Z",
+      "host":"localhost",
+      "path":"/path/to/logstash-tutorial-dataset",
+      "clientip":"83.149.9.216",
+      "ident":"-",
+      "auth":"-",
+      "timestamp":"04/Jan/2015:05:13:45 +0000",
+      "verb":"GET",
+      "request":"/presentations/logstash-monitorama-2013/images/frontend-response-codes.png",
+      httpversion":"1.1",
+      "response":"200",
+      "bytes":"52878",
+      "referrer":"\"http://semicomplete.com/presentations/logstash-monitorama-2013/\"",
+      "agent":"\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/32.0.1700.77 Safari/537.36\""
+      }
+    }]
+  }
+}
+
+Try another search for the geographic information derived from the IP address:
+
+[source,shell]
+curl -XGET 'localhost:9200/logstash-2015.07.30/_search?q=response=401'
+
+[float]
+[[multiple-input-output-plugins]]
+=== Multiple Input and Output Plugins
+
+The information you need to manage often comes from several disparate sources, and use cases can require multiple 
+destinations for your data. Your Logstash pipeline can use multiple input and output plugins to handle these 
+requirements.
+
+This example creates a Logstash pipeline that takes input from a Twitter feed and the Logstash Forwarder client, then 
+sends the information to an Elasticsearch cluster as well as writing the information directly to a file.
+
+[float]
+[[twitter-configuration]]
+==== Reading from a Twitter feed
+
+To add a Twitter feed, you need several pieces of information:
+
+* A _consumer_ key, which uniquely identifies your Twitter app, which is Logstash in this case.
+* A _consumer secret_, which serves as the password for your Twitter app.
+* One or more _keywords_ to search in the incoming feed.
+* An _oauth token_, which identifies the Twitter account using this app.
+* An _oauth token secret_, which serves as the password of the Twitter account.
+
+Visit https://dev.twitter.com/apps to set up a Twitter account and generate your consumer key and secret, as well as 
+your OAuth token and secret.
+
+Use this information to add the following lines to the input section of the `first-pipeline.conf` file:
+
+[source,json]
+twitter {
+    consumer_key =>
+    consumer_secret =>
+    keywords =>
+    oauth_token =>
+    oauth_token_secret => 
+}
+
+[float]
+[[configuring-lsf]]
+==== The Logstash Forwarder
+The Logstash Forwarder is a software client that processes data at the machine of origin before sending the data to the 
+Logstash instance. Data from the Logstash Forwarder uses a secure protocol called _lumberjack_ that is designed for 
+reliability and low latency. Default Logstash configuration includes the lumberjack input plugin, which is designed to 
```

Comment:
Isn't it more the LF that's designed to be resource-friendly? I'd lead the info about the Lumberjack plugin with something like, "The Logstash Lumberjack plugin ingests the data sent by the Logstash Forwarder."